optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/valuerange/buildin/temporal/TemporalValueRange.java

```diff
@@ -45,35 +63,43 @@ public TemporalValueRange(Temporal_ from, Temporal_ to, long incrementUnitAmount
         if (!from.isSupported(incrementUnitType) || !to.isSupported(incrementUnitType)) {
             throw new IllegalArgumentException("The " + getClass().getSimpleName()
                     + " must have a incrementUnitType (" + incrementUnitType
-                    + ") that is supported by its from (" + from + ") and to (" + to + ").");
+                    + ") that is supported by its from (" + from.getClass().getSimpleName()
+                    + ") and to (" + to.getClass().getSimpleName() + ").");
```

Comment:
I still think it wouldn't hurt to **also** show the from and to values too in the error message.